frontend/src/bundles/common/components/protected-route/protected-route.tsx

```diff
@@ -3,18 +3,18 @@ import { type ReactNode } from 'react';
 import { Navigate } from '~/bundles/common/components/components.js';
 import { AppRoute } from '~/bundles/common/enums/enums.js';
 import { useAppSelector } from '~/bundles/common/hooks/hooks.js';
-import { type AsyncThunkConfig } from '~/bundles/common/types/types.js';
 
 type ProtectedRouteProperties = {
     children: ReactNode;
 };
 
 const ProtectedRoute: React.FC<ProtectedRouteProperties> = ({ children }) => {
-    const userAuthenticated = useAppSelector(
-        (state: AsyncThunkConfig['state']) => state.auth.user,
-    );
+    const { isRefreshing, userAuthenticated } = useAppSelector(({ auth }) => ({
+        isRefreshing: auth.isRefreshing,
```

Comment:
why do we need this flag?

Reply:
We need it not to redirect to the sign-in page while the user is refreshing